src/image.rs

```diff
@@ -26,14 +35,30 @@ fn default_context() -> String {
 pub enum Image {
     Name(String),
     Build { build: BuildSpec },
+    Yaml { yaml: YamlSpec },
 }
 
 impl Image {
     /// Name of the image
-    pub fn name(&self) -> String {
+    pub fn name(&self) -> Result<String, Error> {
         match *self {
-            Image::Name(ref s) => s.clone(),
-            Image::Build { ref build } => build.name.clone() + ":floki",
+            Image::Name(ref s) => Ok(s.clone()),
+            Image::Build { ref build } => Ok(build.name.clone() + ":floki"),
+            Image::Yaml { ref yaml } => {
+                let contents = fs::read_to_string(&yaml.file)?;
+                let raw = YamlLoader::load_from_str(&contents)?;
+                let path = yaml.key.split('.').collect::<Vec<_>>();
+                let mut val = &raw[0];
+                for key in &path {
+                    val = match key.parse::<usize>() {
```

Comment:
This is somewhat mysterious - I'm a little puzzled as to how this match is working - care to explain?

Reply:
(probably will become clear on reading the docs)

Reply:
Added a comment - does that explain it?

Reply:
I think I just find the API confusing - let me look into it, and I''ll figure out what I want the comment to be. Happy if it works for now :-)